src/SAML2/SOAP.php

```diff
@@ -7,6 +7,9 @@
  */
 class SAML2_SOAP extends SAML2_Binding
 {
+
+    public $AssertionConsumerServiceURL;
```

Comment:
This is wrong, and derived from the confusion with the constant identifying the binding that I was commenting on. Every class that extends `SAML2_Binding` has a `location` property that stores the location where the message is being sent.

Adding an `$AssertionConsumerServiceURL` property here has two implications:

* We open up for bugs, like the one we have now that leads to the client to raise an exception, because the `AssertionConsumerServiceURL` is set to an empty string and that doesn't match the real ACS URL.
* We are assuming that this class is only suitable for SAML responses, since those are the ones that could carry an ACS URL, but the class is a generic `SAML2_SOAP` that doesn't match with either what it does or how it is used (i.e. the `BINDING_SOAP_RESPONSE` constant.